File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.cpp
class Solution {
public:
    
    
    int findDuplicate(vector<int>& nums) {
        
        for(int i=0;i<nums.size();i++){
            
            int curr=abs(nums[i]);
            if(nums[curr]<0){
                    return curr;
            }
            nums[curr]*=-1;
            
        }
        
        for (auto& num : nums)
            cout<<num<<"\t";
        
        return -1;
    }
};

File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        
            for item in nums:
                
          
                val=abs(item)
                if nums[val]<0:
                        return val
                    
                nums[val]*=-1
                
                        
            
            return -1
        